homeassistant/components/rointe/config_flow.py

```diff
@@ -0,0 +1,121 @@
+"""Config flow for Rointe Heaters integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from rointesdk.rointe_api import RointeAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import CONF_INSTALLATION, CONF_PASSWORD, CONF_USERNAME, DOMAIN, LOGGER
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for Rointe Heaters."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Config flow init."""
+        self.step_user_data: dict[str, Any] | None = None
+        self.step_user_local_id: str | None = None
+        self.step_user_installations: dict[str, Any] | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Step: User credentials validation."""
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        rointe_api = RointeAPI(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+
+        login_error_code = await self.hass.async_add_executor_job(
+            rointe_api.initialize_authentication
+        )
+
+        if not login_error_code.success or not rointe_api.is_logged_in():
+            LOGGER.error(
+                "Error during authentication: %s", login_error_code.error_message
+            )
+            return self.async_show_form(
+                step_id="user",
+                data_schema=STEP_USER_DATA_SCHEMA,
+                errors={"base": "invalid_auth"},
+            )
+
+        installations_response = await self.hass.async_add_executor_job(
+            rointe_api.get_installations
+        )
+
+        if not installations_response.success:
+            LOGGER.error(
+                "Error retrieving installations: %s",
+                installations_response.error_message,
+            )
+
+            return self.async_show_form(
+                step_id="user",
+                data_schema=STEP_USER_DATA_SCHEMA,
+                errors={"base": "unable_get_installations"},
+            )
+
+        installations = installations_response.data
+
+        # If we get this far then we have logged in and determined the local_id. Go the next step.
+        self.step_user_data = user_input
+        self.step_user_installations = installations
+
+        return await self.async_step_installation(None)
+
+    async def async_step_installation(
+        self, user_input: dict[str, Any] | None
+    ) -> FlowResult:
```

Comment:
Okay so apparently 1 account can lead to more installations? Can you maybe elaborate on what an installation is? Usually we have the guideline that 1 (cloud) account = 1 config entry.

Reply:
Essentially it represents a house/home. It's part of their hierarchy: Installations -> Zones ->  Devices.

During configuration the user must pick one installation (for the majority of cases there will only be one option available). The Installation ID is then stored as it is required to retrieve devices.